src/blueapi/service/main.py

```diff
@@ -372,34 +382,39 @@ def start(config: ApplicationConfig):
         "%(asctime)s %(levelprefix)s %(client_addr)s"
         + " - '%(request_line)s' %(status_code)s"
     )
+    app = get_app()
+
     FastAPIInstrumentor().instrument_app(
         app,
         tracer_provider=get_tracer_provider(),
         http_capture_headers_server_request=[",*"],
         http_capture_headers_server_response=[",*"],
     )
     app.state.config = config
+
     uvicorn.run(app, host=config.api.host, port=config.api.port)
 
 
-@app.middleware("http")
-async def add_api_version_header(request: Request, call_next):
-    response = await call_next(request)
-    response.headers["X-API-Version"] = REST_API_VERSION
-    return response
+def add_api_version_header(app: FastAPI):
+    @app.middleware("http")
+    async def add_api_version_header(request: Request, call_next):
+        response = await call_next(request)
+        response.headers["X-API-Version"] = REST_API_VERSION
+        return response
 
 
-@app.middleware("http")
-async def inject_propagated_observability_context(
-    request: Request, call_next: Callable[[Request], Awaitable[Response]]
-) -> Response:
-    """Middleware to extract the any prorpagated observability context from the
-    HTTP headers and attatch it to the local one.
-    """
-    if CONTEXT_HEADER in request.headers:
-        ctx = get_global_textmap().extract(
-            {CONTEXT_HEADER: request.headers[CONTEXT_HEADER]}
-        )
-        attach(ctx)
-    response = await call_next(request)
-    return response
+def inject_propagated_observability_context(app: FastAPI):
+    @app.middleware("http")
+    async def inject_propagated_observability_context(
+        request: Request, call_next: Callable[[Request], Awaitable[Response]]
+    ) -> Response:
+        """Middleware to extract the any propagated observability context from the
+        HTTP headers and attach it to the local one.
+        """
+        if CONTEXT_HEADER in request.headers:
+            ctx = get_global_textmap().extract(
+                {CONTEXT_HEADER: request.headers[CONTEXT_HEADER]}
+            )
+            attach(ctx)
+        response = await call_next(request)
+        return response
```

Comment:
I'd like a better way of passing these into the app, but using app.add_middleware expects a callable that takes app, not request, callable[[request, response]].